### DISCUSSION TITLE
Simplest Next.js / Pocketbase OAuth2 Tutorial

### DISCUSSION THREAD

**Discussion by skellgreco** [pocketbase/pocketbase#5923] [Show and tell]
Feel free to make pull requests to improve this code! This is the simplest way I managed to make them 2 work! Take a look:

https://github.com/skellgreco/pocketbase-oauth2-nextjs

*Edit #1: Currently affected by a vulnerability. Looking into it for a fix!*

**Comment by ganigeorgiev** [comment]
_As mentioned in the reply to your other comment, please make sure to check https://github.com/pocketbase/pocketbase/discussions/5313 because the above template is affected by "Problem 1" described in the discussion._

**Reply by skellgreco** [reply]
Alright! Thanks for reporting! Looking into it to push a fix!

**Reply by skellgreco** [reply]
Hey! Wouldn't just including
` const pb = new PocketBase(pocketBaseURL); `
inside each function fix this vulnairability (as it will create a new connection for each request)? 
Thanks!

**Reply by ganigeorgiev** [reply]
`const pb = new PocketBase(pocketBaseURL);` doesn't send any requests. It just instantiates an object. As long as that object is request scoped then it should be fine.

**Reply by skellgreco** [reply]
I mean the best way to keep it simple while fixing the issue would be to probably just repeat the chunk of code that intializes the pocketbase connection instead of importing it as a component. Seems quite weird but other alternatives would make the template over complicated

**Comment by skellgreco** [comment]
Has the new pocketbase version (0.23) made any changes that could make this code safer?
Thanks

**Reply by ganigeorgiev** [reply]
No, there are no changes related to the above mentioned JS SSR issues, other than [being mentioned in the docs](https://pocketbase.io/docs/client-side-sdks/) (also the above overwrite issue is not really PocketBase specific and could happen with any global object/singleton).

**Reply by ganigeorgiev** [reply]
For the JS SDK changes, you can also refer to its [CHANGELOG](https://github.com/pocketbase/js-sdk/blob/master/CHANGELOG.md#0220).

**Reply by skellgreco** [reply]
Well as far as I understand, my approach can't be really implemented with safety. The only way to properly do it is by having just the admin pocketbase instance manage the data in/out the database. However, if I use that approach, how could the new ratelimits limit just one user and not everyone on the webserver. Wouldn't I have to build a custom ratelimiting system?

**Reply by ganigeorgiev** [reply]
I don't understand what do you mean by _"implemented with safety"_ but I have no energy to discuss it further. If you haven't already, please read https://github.com/pocketbase/pocketbase/discussions/5313.

About the ratelimiter - it really depends how you want to use PocketBase (client-side, server-side, mixed).

If PocketBase is accessible only by the node server then you can enable the "Trusted proxy headers" option and forward a special header with the user IP with each request (you must ensure that only the Node.js server can set it; the header could be attached globally with the [JS SDK `beforeSend` hook](https://github.com/pocketbase/js-sdk?tab=readme-ov-file#send-hooks)).

Of course you can always setup a rate limiter on reverse proxy level if the above doesn't suffice.
